### PR TITLE
Fix timespread for zero-length pixelsets, add tests, fixes #372

### LINKED ISSUE #372: ArrayStoreException in Utils.arrayMultiplication for 20131104_196
Processing of 20131104_196 throws an error in TimeSpread after a couple of hundred events:

```
java.lang.ArrayStoreException: Array of length zero.
	at fact.Utils.arrayMultiplication(Utils.java:418)
	at fact.features.TimeSpread.process(TimeSpread.java:50)
	at stream.runtime.AbstractProcess.process(AbstractProcess.java:127)
	at stream.runtime.AbstractProcess.execute(AbstractProcess.java:185)
	at stream.runtime.ProcessThread.run(ProcessThread.java:129)
Exception in thread "Thread-2" java.lang.RuntimeException: Process stopped after exception: Array of length zero.
	at stream.runtime.ProcessThread.run(ProcessThread.java:171)
java.lang.ArrayStoreException: Array of length zero.
	at fact.Utils.arrayMultiplication(Utils.java:418)
	at fact.features.TimeSpread.process(TimeSpread.java:50)
	at stream.runtime.AbstractProcess.process(AbstractProcess.java:127)
	at stream.runtime.AbstractProcess.execute(AbstractProcess.java:185)
	at stream.runtime.ProcessThread.run(ProcessThread.java:129)
streams exited on error: java.lang.ArrayStoreException: Array of length zero.
java.lang.RuntimeException: java.lang.ArrayStoreException: Array of length zero.
	at stream.runtime.ProcessContainer.run(ProcessContainer.java:580)
	at stream.run.main(run.java:168)
	at stream.run.main(run.java:147)
	at fact.run.main(run.java:26)
Caused by: java.lang.ArrayStoreException: Array of length zero.
	at fact.Utils.arrayMultiplication(Utils.java:418)
	at fact.features.TimeSpread.process(TimeSpread.java:50)
	at stream.runtime.AbstractProcess.process(AbstractProcess.java:127)
	at stream.runtime.AbstractProcess.execute(AbstractProcess.java:185)
	at stream.runtime.ProcessThread.run(ProcessThread.java:129)
```

### PR DESCRIPTION
**Pull request by MaxNoe**

### CODE REVIEW

File: src/main/java/fact/features/TimeSpread.java [hunk 1]
```diff
@@ -27,36 +29,22 @@ public Data process(Data item) {
...
-        // Sum over the weight array:
-        double sumw = Utils.arraySum(w);
-        // Sum over the weighted time array:
-        double sumwt = Utils.arraySum(Utils.arrayMultiplication(w, t));
-        // Sum over the element wise multiplication of t and t:
-        double sumtt = Utils.arraySum(Utils.arrayMultiplication(t, t));
-        // Sum over the element wise multiplication of t and t, weighted:
-        double sumwtt = Utils.arraySum(Utils.arrayMultiplication(w, Utils.arrayMultiplication(t, t)));
+        double[] t = shower.stream().mapToDouble(p -> arrivalTime[p.id]).toArray();
+        double[] w = shower.stream().mapToDouble(p -> weights[p.id]).toArray();
```
Comment (jebuss) [inline src/main/java/fact/features/TimeSpread.java]:
```suggestion
        double[] w = shower.stream().mapToDouble(p -> weights[p.id]).filter(w -> w >= 0).toArray();
```

File: src/main/java/fact/features/TimeSpread.java [hunk 2]
```diff
@@ -27,36 +29,22 @@ public Data process(Data item) {
...
-        double sumt = Utils.arraySum(t);
-        // Sum over the weight array:
-        double sumw = Utils.arraySum(w);
-        // Sum over the weighted time array:
-        double sumwt = Utils.arraySum(Utils.arrayMultiplication(w, t));
-        // Sum over the element wise multiplication of t and t:
-        double sumtt = Utils.arraySum(Utils.arrayMultiplication(t, t));
-        // Sum over the element wise multiplication of t and t, weighted:
-        double sumwtt = Utils.arraySum(Utils.arrayMultiplication(w, Utils.arrayMultiplication(t, t)));
+        double[] t = shower.stream().mapToDouble(p -> arrivalTime[p.id]).toArray();
```
Comment (jebuss) [inline src/main/java/fact/features/TimeSpread.java]:
```suggestion
        double[] t = shower.stream().filter(p -> weights[p.id] >= 0).mapToDouble(p -> arrivalTime[p.id]).toArray();
```

File: src/main/java/fact/features/TimeSpread.java [hunk 3]
```diff
@@ -27,36 +29,22 @@ public Data process(Data item) {
...
 
+        Variance var = new Variance();
+        var.setBiasCorrected(false);
 
-        double timespread = Math.sqrt(sumtt / n - Math.pow(sumt / n, 2));
-        double weightedTimespread = Math.sqrt(sumwtt / sumw - Math.pow(sumwt / sumw, 2));
-
+        double timespread = Math.sqrt(var.evaluate(t));
+        double weightedTimespread;
+        if (shower.size() > 0) {
```
Comment (jebuss) [inline src/main/java/fact/features/TimeSpread.java]:
```suggestion
        if (t.length > 0) {
```

**Comment by MaxNoe** [comment]
Now ignoring negative weights instead of not calculating it on the pedestals

**Pull request merged**